### ISSUE TITLE
wrong current directory in workspace

### ISSUE THREAD

**Issue by andrewbanchich** [passcod/cargo-watch#172]
I just did a `cargo install cargo-watch` to update and running `cargo watch -c -x "run"` from a workspace directory will run the workspaces `default-members` binary instead.

Workspace `Cargo.toml`:
```
[workspace]
default-members = ["foo"]

members = [
    "foo",
    "bar/baz",
]
```

If I `cd` into `./bar/baz` and run `cargo watch -c -x "run"`, `foo` will run. If I run `cargo run` from the same directory, `baz` runs.

**Comment by andrewbanchich** [comment]
Downgrading to 7.7.0 works. Looks like 7.7.1 introduced the change.

**Comment by passcod** [comment]
Mmm. I'm not sure which way is the expected behaviour.

Context: the behaviour change is because the code that changes the directory was looking for the first parent with a `Cargo.toml`, and that got changed to read the manifests and find the _real_ cargo root instead.

However, when the code was originally written, workspaces weren't a thing, and the _intent_ was indeed to run from the root. The code was literally named `fn cargo_root()`.

I think remaking the _effective_ behaviour is probably the right thing to do.

**Comment by passcod** [comment]
Fixes in 7.8.0

**Comment by andrewbanchich** [comment]
Awesome, thanks!
